lib/directory-tree.js

```diff
@@ -27,16 +27,18 @@ function directoryTree (path, options, onEachFile) {
 
 	try { stats = FS.statSync(path); }
 	catch (e) { return null; }
-	if (stats.isFile()) {
-		const ext = PATH.extname(path).toLowerCase();
 
-		// Only add files with the provided extensions
-		if (options && 
-			options.extensions && 
-			options.extensions.length && 
-			options.extensions.indexOf(ext) === -1)
-			return null;
+	// Skip if it matches the exclude regex
+	if (options && options.exclude && options.exclude.test(path))
+		return null;  
 
+	if (stats.isFile()) {
+		// Skip if it does not match the include regex
+		if (options && options.includeOnly && !options.includeOnly.test(path))
```

Comment:
Yes, I had initially kept it outside. But I realized that I will need to rework most of the tests to get them to pass. We can't check for extensions anymore in most tests because the directory won't match the regex. The regex will need to be something like `/\.txt$|test_data/`. Or we can skip this option completely in the test. The only problem with skipping it, is that .DS_Store will get included for mac users. I'm in favor of skipping the options in some tests, because the test should only test one thing.